Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2421,6 +2421,7 @@ extension BrowserViewController: Themeable {
         guard let contentScript = self.tabManager.selectedTab?.getContentScript(name: ReaderMode.name()) else { return }
         appyThemeForPreferences(profile.prefs, contentScript: contentScript)
 
+        overrideUserInterfaceStyle = ThemeManager.instance.userInterfaceStyle
```

Comment:
Great find and I think this seals the deal for now. We need to revisit it after the next upgrade probably as by then we'll have the solution to set `overrideUserInterfaceStyle ` on the window and not on VC level